### PR TITLE
New machines marked as NOT_WORKING

### PR DESCRIPTION
**Pull request by wilbertpol**
----------------------------------
Atari 2600 Point of Purchase Display Unit [Wilbert Pol]

### CODE REVIEW

**Comment by wilbertpol** [comment]
I just realised that it would be more correct to use [historicalsource, Wilbert Pol] for the credits

File: src/mame/drivers/a2600.cpp [hunk 1]
```diff
@@ -42,6 +114,75 @@ void a2600_base_state::a2600_mem(address_map &map) // 6507 has 13-bit address sp
...
+{
+	map(0x0000, 0x007f).mirror(0x0f00).rw(m_tia, FUNC(tia_video_device::read), FUNC(tia_video_device::write));
+	map(0x0080, 0x00ff).mirror(0x0d00).ram().share("riot_ram");
+#if USE_NEW_RIOT
+	map(0x0280, 0x029f).mirror(0x0d00).m("riot", FUNC(mos6532_t::io_map));
+#else
+	map(0x0280, 0x029f).mirror(0x0d00).rw("riot", FUNC(riot6532_device::read), FUNC(riot6532_device::write));
+#endif
+	map(0x0800, 0x0800).rw(FUNC(a2600_pop_state::rom_switch_r), FUNC(a2600_pop_state::rom_switch_w));
+	map(0x1000, 0x1fff).rw(m_bankdev, FUNC(address_map_bank_device::read8), FUNC(address_map_bank_device::write8));
```
Comment (cuavas) [inline src/mame/drivers/a2600.cpp:127]: Views are preferred over going through multiple levels of indirection and trampolines for `address_map_bank_device`.

File: src/mame/drivers/a2600.cpp [hunk 2]
```diff
@@ -42,6 +114,75 @@ void a2600_base_state::a2600_mem(address_map &map) // 6507 has 13-bit address sp
...
+}
+
+
+void a2600_pop_state::banked_map(address_map &map)
+{
+	map(0x00000, 0x02ffff).rom().region("maincpu", 0);
+}
```
Comment (cuavas) [inline src/mame/drivers/a2600.cpp:134]: Why are you using a bank device for plain ROM anyway?  Can’t you just use a plain memory bank?

File: src/mame/drivers/a2600.cpp [hunk 3]
```diff
@@ -42,6 +114,75 @@ void a2600_base_state::a2600_mem(address_map &map) // 6507 has 13-bit address sp
...
+		} else if (a8 & 0x04) {
+			// 30 seconds
+			reset_time = attotime::from_seconds(30);
+		} else if (a8 & 0x02) {
+			// 1 minute
+			reset_time = attotime::from_seconds(60);
+		} else if (a8 & 0x01) {
+			// 2 minutes
+			reset_time = attotime::from_seconds(120);
+		}
```
Comment (cuavas) [inline src/mame/drivers/a2600.cpp:166]: The rest of this file is Allman style.

File: src/mame/drivers/a2600.cpp [hunk 4]
```diff
@@ -502,12 +685,136 @@ void a2600_state::a2600p(machine_config &config)
...
+	m_riot->out_pa_callback().set(FUNC(a2600_pop_state::switch_A_w));
+	m_riot->in_pb_callback().set_ioport("SWB");
+	m_riot->out_pb_callback().set(FUNC(a2600_pop_state::switch_B_w));
+	m_riot->irq_callback().set(FUNC(a2600_pop_state::irq_callback));
+#endif
+
+	VCS_CONTROL_PORT(config, CONTROL1_TAG, vcs_control_port_devices, "joy");
+	VCS_CONTROL_PORT(config, CONTROL2_TAG, vcs_control_port_devices, "joy");
+
+	ADDRESS_MAP_BANK(config, "bank").set_map(&a2600_pop_state::banked_map).set_options(ENDIANNESS_LITTLE, 8, 22, 0x1000);
```
Comment (cuavas) [inline src/mame/drivers/a2600.cpp:728]: Is there really nothing in common with the home console that can be brought out to a common function?

File: src/mame/drivers/a2600.cpp [hunk 5]
```diff
@@ -502,12 +685,136 @@ void a2600_state::a2600p(machine_config &config)
...
+
+
 /*    YEAR  NAME    PARENT  COMPAT  MACHINE  INPUT  CLASS        INIT        COMPANY     FULLNAME */
 CONS( 1977, a2600,  0,      0,      a2600,   a2600, a2600_state, empty_init, "Atari",    "Atari 2600 (NTSC)" , MACHINE_SUPPORTS_SAVE )
 CONS( 1978, a2600p, a2600,  0,      a2600p,  a2600, a2600_state, empty_init, "Atari",    "Atari 2600 (PAL)",   MACHINE_SUPPORTS_SAVE )
+
+// Released in 1981/1982
+// Games 35-42 are copyright 1982 and looking at the game list they seem to be
+// added later.
+CONS( 198?, a2600_pop, 0,      0,      a2600_pop, a2600_pop, a2600_pop_state, empty_init, "Atari",    "Atari 2600 Point of Purchase Display",   MACHINE_NOT_WORKING | MACHINE_SUPPORTS_SAVE )
```
Comment (cuavas) [inline src/mame/drivers/a2600.cpp:820]: This system is for public use – it should be `GAME`.

**Review by cuavas — CHANGES_REQUESTED** [review]

File: src/mame/drivers/a2600.cpp (hunk 1 above)
Comment (wilbertpol) [reply src/mame/drivers/a2600.cpp]: changed to memory bank

File: src/mame/drivers/a2600.cpp (hunk 2 above)
Comment (wilbertpol) [reply src/mame/drivers/a2600.cpp:134]: changed to memory bank

File: src/mame/drivers/a2600.cpp (hunk 5 above)
Comment (wilbertpol) [reply src/mame/drivers/a2600.cpp]: changed to GAME

File: src/mame/drivers/a2600.cpp (hunk 3 above)
Comment (wilbertpol) [reply src/mame/drivers/a2600.cpp]: Code style now matches the rest of the file

File: src/mame/drivers/a2600.cpp (hunk 4 above)
Comment (wilbertpol) [reply src/mame/drivers/a2600.cpp]: Moved the common parts to a2600_base_ntsc

**Pull request merged**